liborbum/src/Controller/Ee/Vpu/Vif/CVif.cpp

```diff
@@ -95,48 +95,101 @@ void CVif::INSTRUCTION_UNSUPPORTED(VifUnit_Base* unit, const VifcodeInstruction
     throw std::runtime_error("VIFcode CMD field was invalid! Please fix.");
 }
 
+// Refer to EE Users Manual pg 103.
 void CVif::NOP(VifUnit_Base* unit, const VifcodeInstruction inst)
 {
+    // nothing to do
 }
 
+// Refer to EE Users Manual pg 104.
 void CVif::STCYCL(VifUnit_Base* unit, const VifcodeInstruction inst)
 {
+    // Writes CODE.IMMEDIATE to CYCLE
+    uword immediate = unit->code.IMMEDIATE.extract_from(unit->code.read_uword());
+    unit->cycle.write_uword(immediate);
 }
 
+// Refer to EE Users Manual pg 105.
 void CVif::OFFSET(VifUnit_Base* unit, const VifcodeInstruction inst)
 {
+    // VIF1 only
+    if (unit->core_id != 1)
+        return;
+
+    // Clear STAT.DBF
+    uword status = 0;
+    unit->stat.write_uword(unit->stat.DBF.insert_into(unit->stat.read_uword(), status));
```

Comment:
To insert a bitfield value, you can just use `unit->stat.insert_field(VifUnitRegister_Stat::DBF, 0);`

The `insert_field` (write) and `extract_field` (read) methods should be used when dealing with registers when possible.